tests/framework/extensions/provisioning/verify.go

```diff
@@ -335,3 +344,43 @@ func GetSnapshots(client *rancher.Client, localclusterID string, clusterName str
 	return snapshots, nil
 
 }
+
+// VerifySSHTests validates the ssh tests listed in the config on each node of the cluster
+func VerifySSHTests(t *testing.T, client *rancher.Client, clusterObject *steveV1.SteveAPIObject, sshTests []string, clusterID string) {
+	client, err := client.ReLogin()
+	require.NoError(t, err)
+
+	clusterSpec := &provv1.ClusterSpec{}
+	err = steveV1.ConvertToK8sType(clusterObject.Spec, clusterSpec)
+	require.NoError(t, err)
+
+	steveclient, err := client.Steve.ProxyDownstream(clusterID)
+	require.NoError(t, err)
+	nodesSteveObjList, err := steveclient.SteveType("node").List(nil)
+	require.NoError(t, err)
+
+	dynamicSchema := clusterSpec.RKEConfig.MachinePools[0].DynamicSchemaSpec
+	var data clusters.DynamicSchemaSpec
+	err = json.Unmarshal([]byte(dynamicSchema), &data)
+	require.NoError(t, err)
+
+	sshUser := data.ResourceFields.SSHUser.Default.StringValue
+	for _, tests := range sshTests {
+		for _, rancherNode := range nodesSteveObjList.Data {
```

Comment:
```suggestion
    for _, node := range nodesSteveObjList.Data {
```

This is not a rancher node, but rather a corev1.Node object in the downstream cluster, aka a k8s node, calling it a rancherNode makes it sounds confusing.